sapi/fpm/fpm/fpm_unix.c

```diff
@@ -349,6 +358,102 @@ static int fpm_unix_conf_wp(struct fpm_worker_pool_s *wp) /* {{{ */
 }
 /* }}} */
 
+#if HAVE_FPM_CPUAFFINITY
+struct fpm_cpuaffinity_conf {
+#if defined(HAVE_SCHED_SETAFFINITY) || defined(HAVE_CPUSET_SETAFFINITY)
+	cpu_set_t cset;
+#endif
+	long min;
+	long max;
+};
+
+static long fpm_cpumax(void)
+{
+	static long cpuid = LONG_MIN;
+	if (cpuid == LONG_MIN) {
+		cpuid = sysconf(_SC_NPROCESSORS_ONLN);
+	}
+
+	return cpuid;
+}
+
+static void fpm_cpuaffinity_init(struct fpm_cpuaffinity_conf *c)
+{
+#if defined(HAVE_SCHED_SETAFFINITY) || defined(HAVE_CPUSET_SETAFFINITY)
+	CPU_ZERO(&c->cset);
+#endif
+}
+
+static void fpm_cpuaffinity_add(struct fpm_cpuaffinity_conf *c)
+{
+#if defined(HAVE_SCHED_SETAFFINITY) || defined(HAVE_CPUSET_SETAFFINITY)
+	int i;
+
+	for (i = c->min; i <= c->max; i ++) {
+		if (!CPU_ISSET(i, &c->cset)) {
+			CPU_SET(i, &c->cset);
+		}
+	}
+#endif
+}
+
+static int fpm_cpuaffinity_set(struct fpm_cpuaffinity_conf *c)
+{
+#if defined(HAVE_SCHED_SETAFFINITY)
+	return sched_setaffinity(0, sizeof(c->cset), &c->cset);
+#elif defined(HAVE_CPUSET_SETAFFINITY)
+	return cpuset_setaffinity(CPU_LEVEL_WHICH, CPU_WHICH_PID, -1, sizeof(c->cset), &c->cset);
+#endif
+}
+
+static void fpm_cpuaffinity_destroy(struct fpm_cpuaffinity_conf *c)
+{
+	// some platform/apis require to allocates data on the heap
+}
```

Comment:
it will be useful for netbsd portage for example

Reply:
It would be better to either implement for all platforms. Ideally I would prefer not to have merged anything that needs some further work or is only useful for some potential future work.